src/core/ngx_memcache.c

```diff
@@ -1984,6 +2005,27 @@ static ngx_uint_t ngx_memcache_perl_hash (u_char *key, size_t len)
     return h;
 }
 
+static u_char* ngx_sha256_hash (u_char* key, size_t len)
+{
+    u_char digest[SHA256_KEY_LENGTH];
+    SHA256(key, len, digest);
+
+    /*
+     * this will give you back an integer representation of the hash,
+     * you can then turn this into a hex representation,
+     * if you want to pass it around as a string.
+     */
+
+    static u_char hashed_key[SHA256_KEY_LENGTH+1];
```

Comment:
I would make this variable dynamically allocated on the heap just to be sure. I'm not sure if this is an actual issue but using a static variable (essentially a global) might lead to race conditions.

Reply:
Done